src/devices/Bmp180/Bmp180.cs

```diff
@@ -0,0 +1,261 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Ported from https://github.com/adafruit/Adafruit_Python_BMP/blob/master/Adafruit_BMP/BMP085.py  
+// Formulas and code examples can also be found in the datasheet https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+
+using System;
+using System.Device.I2c;
+using System.Buffers.Binary;
+using System.Threading;
+using Iot.Units;
+
+namespace Iot.Device.Bmp180
+{
+    public class Bmp180 : IDisposable
+    {
+        private I2cDevice _i2cDevice;        
+        private readonly CalibrationData _calibrationData;
+        private Sampling _mode;
+        public const byte DefaultI2cAddress = 0x77;
+
+        public Bmp180(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;            
+            _calibrationData = new CalibrationData();            
+            //Read the coefficients table
+            _calibrationData.ReadFromDevice(this);
+            SetSampling(Sampling.Standard);
+        }
+        
+        /// <summary>
+        /// Sets sampling to the given value
+        /// </summary>
+        /// <param name="mode">Sampling Mode</param>
+        public void SetSampling(Sampling mode)
+        {
+            _mode = mode;
+        }
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public Temperature ReadTemperature()
+        {
+            //Gets the compensated temperature in degrees celsius
+            short UT = ReadRawTemperature();
+
+            // Calculations below are taken straight from section 3.5 of the datasheet.
+            int X1 = (UT - _calibrationData.AC6) * _calibrationData.AC5 / (1 << 15);
+            int X2 = _calibrationData.MC * (1 << 11) / (X1 + _calibrationData.MD);
+            int B5 = (X1 + X2);
+            int T = (B5 + 8) / (1 << 4);
+            
+            return Temperature.FromCelsius((double)T/10);            
+        }
+
+        /// <summary>
+        ///  Reads the pressure from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Atmospheric pressure in Pa
+        /// </returns>
+        public int ReadPressure()
+        {
+            //Gets the compensated pressure in Pascals
+            short UT = ReadRawTemperature();
+            int UP = ReadRawPressure();
+
+            // Calculations below are taken straight from section 3.5 of the datasheet.
+            // Calculate true temperature coefficient B5.

```

Comment:
since you're duplicating some logic here it might be worth making a function out of that